infrastructure/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java

```diff
@@ -0,0 +1,986 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * This class has test methods that test the RPC calls present in {@link SequencerServer}.
+ * The RequestMsg type messages that are tested are as follows -
+ * - TokenRequestMsg
+ * - BootstrapSequencerRequestMsg
+ * - SequencerTrimRequestMsg
+ * - SequencerMetricsRequestMsg
+ * - StreamsAddressRequestMsg
+ * See test methods for their functionality descriptions.
+ */
+@Slf4j
+public class SequencerServerTest {
+
+    private final AtomicInteger requestCounter = new AtomicInteger();
+
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+
+    // The SequencerServer instance used for testing
+    private SequencerServer sequencerServer;
+
+    // Parameters that need to be mocked
+    @Mock
+    private ServerContext mockServerContext;
+    @Mock
+    private IServerRouter mockServerRouter;
+    @Mock
+    private ChannelHandlerContext mockChannelHandlerContext;
+    @Mock
+    private Map<UUID, StreamAddressSpace> mockStreamAddressSpaceMap;
+    @Mock
+    private Map<UUID, Long> mockStreamTailToGlobalTailMap;
+
+    // Using spy as we use the default behaviour most of the times and
+    // override only a few times.
+    @Spy
+    private SequencerServer.SequencerFactoryHelper spySequencerFactoryHelper;
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values.
+     *
+     * @param ignoreClusterId indicates if the message is clusterId aware
+     * @param ignoreEpoch     indicates if the message is epoch aware
+     * @return the corresponding HeaderMsg
+     */
+    private HeaderMsg getBasicHeader(boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, 0L,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
+
+    /**
+     * A helper method that compares the base fields of two message headers.
+     * These include the request ID, the epoch, the client ID, and the cluster ID.
+     *
+     * @param requestHeader  the header from the request message
+     * @param responseHeader the header from the response message
+     * @return true if the two headers have the same base field values
+     */
+    private boolean compareBaseHeaderFields(HeaderMsg requestHeader, HeaderMsg responseHeader) {
+        return requestHeader.getRequestId() == responseHeader.getRequestId() &&
+                requestHeader.getEpoch() == responseHeader.getEpoch() &&
+                requestHeader.getClientId().equals(responseHeader.getClientId()) &&
+                requestHeader.getClusterId().equals(responseHeader.getClusterId());
+    }
+
+    /**
+     * Perform the required preparation before running individual tests.
+     * This includes preparing the mocks and initializing the DirectExecutorService.
+     */
+    @Before
+    public void setup() {
```

Comment:
Maybe consider factoring some of the common preparation code here? If nothing can be cleanly factored, consider updating the comment.

Reply:
Updated the comment, thanks!